### PR TITLE
Credits cya

### PR DESCRIPTION
**Pull request by mauro-fanton**
### Description of Work carried through

Brief description of what/why/how.

### CODE REVIEW

File: app/controllers/returns/ReturnsCheckYourAnswersController.scala [hunk 1]
```diff
@@ -77,4 +68,42 @@ class ReturnsCheckYourAnswersController @Inject()(
...
+
+        Ok(view(returnViewModel, credits)(request, messages))
+    }
+  }
+
+  private def getCredits(request: DataRequest[_])
+                        (implicit hc: HeaderCarrier): Future[Either[ServiceError, Credits]] = {
+    if(request.userAnswers.getOrFail(WhatDoYouWantToDoPage)) {
+        creditsCalculatorConnector.get(request.pptReference).map {
+          case Right(creditBalance) => Right(CreditsClaimedDetails(request.userAnswers, creditBalance = creditBalance))
```
Comment (liampan) [inline app/controllers/returns/ReturnsCheckYourAnswersController.scala:101]: Does this need to wrap up with user answers? The calling method already has useranswers as it provided the request

File: app/models/returns/CreditsClaimedDetails.scala [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      CreditConvertedAnswerPartialKey -> convertedAnswerYesNoAsString,
+      CreditConvertedWeightPartialKey -> converted.value.asKg,
+      CreditsTotalWeightPartialKey -> totalWeight.asKg,
+      CreditTotalPartialKey -> totalCredits.asPounds
+    ).filter(canShowWeight(_))
+  }
+
+  private def canShowWeight(keyValue: (String,String)): Boolean = {
+    keyValue match {
+      case (CreditExportedWeightPartialKey, _) => exportedAnswerYesNoAsString.equals("site.yes")
```
Comment (liampan) [inline app/models/returns/CreditsClaimedDetails.scala:52]:
```suggestion
      case (CreditExportedWeightPartialKey, _) => exported.yesNo
```

File: app/views/components/returns/checkYourAnswer/credits/CreditsClaimed.scala.html [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+@paragraph(
+    content = link(
+        id = "change-credit-link",
+        text = messages("submit-return.check-your-answers.credits.change.text.link"),
+        call = controllers.returns.credits.routes.ExportedCreditsController.onPageLoad(CheckMode)
+    )
+)
+
+<!-- todo: create YesAndNo page and redirect to this  -->
```
Comment (liampan) [inline app/views/components/returns/checkYourAnswer/credits/CreditsClaimed.scala.html:49]: is this on the work flowy todo list?

File: app/views/components/returns/checkYourAnswer/credits/NoCreditsClaimed.scala.html [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    link: Link
+)
+
+@()(implicit request: Request[_], messages: Messages)
+
+@paragraph(
+    content = Html(messages("submit-return.check-your-answers.credits.not.claimed.hint"))
+)
+
+<!-- todo: add a proper link for the credit change -->
```
Comment (liampan) [inline app/views/components/returns/checkYourAnswer/credits/NoCreditsClaimed.scala.html:29]: is this on the work flowy todo list?

File: test/views/returns/credits/ConvertedCreditsViewSpec.scala [hunk 5]
```diff
@@ -26,7 +26,7 @@ import support.{ViewAssertions, ViewMatchers}
 import views.html.returns.credits.ConvertedCreditsView
 
 class ConvertedCreditsViewSpec extends ViewSpecBase with ViewAssertions with ViewMatchers {
-
+  //todo add tests
```
Comment (liampan) [inline test/views/returns/credits/ConvertedCreditsViewSpec.scala:29]: 👀

File: app/views/components/returns/checkYourAnswer/credits/NoCreditsClaimed.scala.html (hunk 4 above)
Comment (lmcewanhmrc) [reply app/views/components/returns/checkYourAnswer/credits/NoCreditsClaimed.scala.html:29]: add it thanks

File: app/views/components/returns/checkYourAnswer/credits/CreditsClaimed.scala.html (hunk 3 above)
Comment (lmcewanhmrc) [reply app/views/components/returns/checkYourAnswer/credits/CreditsClaimed.scala.html:49]: I think so - if this is the "remove credit claim" link

File: app/views/components/returns/checkYourAnswer/credits/NoCreditsClaimed.scala.html (hunk 4 above)
Comment (lmcewanhmrc) [reply app/views/components/returns/checkYourAnswer/credits/NoCreditsClaimed.scala.html:29]: sorry, "have added it thanks"... can't type this morning

**Review by lmcewanhmrc — APPROVED** [review]

**Pull request merged**